docs/guide/improving-performance.md

```diff
@@ -72,3 +72,102 @@ export default defineConfig({
 })
 ```
 :::
+
+## Sharding
+
+Test sharding means running a small subset of test cases at a time. It can be useful when you have multiple machines that could be used to run tests simultaneously.
+
+To split Vitest tests on multiple different runs, use [`--shard`](/guide/cli#shard) option with [`--reporter=blob`](/guide/reporters#blob-reporter) option:
+
+```sh
+vitest run --reporter=blob --shard=1/3 # 1st machine
+vitest run --reporter=blob --shard=2/3 # 2nd machine
+vitest run --reporter=blob --shard=3/3 # 3rd machine
+```
+
+Collect the results stored in `.vitest-reports` directory from each machine and merge them with [`--merge-reports`](/guide/cli#merge-reports) option:
+
+```sh
+vitest --merge-reports
+```
+
+<details>
+  <summary>Github action example</summary>
+
+```yaml
+# Inspired from https://playwright.dev/docs/test-sharding
+name: Tests
```

Comment:
I haven't yet tested this, but @sheremet-va & @hi-ogawa what do you think about adding such example here?
I was also thinking about adding a complete example to https://github.com/vitest-tests maybe.

Reply:
https://github.com/vitest-tests/test-sharding/actions/runs/9180138101

Reply:
Looks awesome! Having this in the doc and also a link to your example repo should be very useful.